godot-kotlin/godot-library/src/nativeCore/kotlin/godot/core/type/PoolColorArray.kt

```diff
@@ -0,0 +1,187 @@
+     */
+    operator fun get(idx: Int): Color {
+        return Color(
+            callNative {
+                checkNotNull(Godot.gdnative.godot_pool_color_array_get)(it, idx)
+            }
+        )
+    }
+
+    /**
+     * Inserts a new element at a given position in the array.
+     * The position must be valid, or at the end of the array (idx == size()).
+     */
+    fun insert(idx: Int, data: Color) {
+        callNative {
+            checkNotNull(Godot.gdnative.godot_pool_color_array_insert)(it, idx, data.getRawMemory(this).reinterpret())
+        }
+    }
+
+    /**
+     * Reverses the order of the elements in the array.
+     */
+    fun invert() {
+        callNative {
+            checkNotNull(Godot.gdnative.godot_pool_color_array_invert)(it)
+        }
+    }
+
+    /**
+     * Appends a value to the array.
+     */
+    fun pushBack(data: Color) {
+        callNative {
+            checkNotNull(Godot.gdnative.godot_pool_color_array_push_back)(it, data.getRawMemory(this).reinterpret())
+        }
+    }
+
+    /**
+     * Removes an element from the array by index.
+     */
+    fun remove(idx: Int) {
+        callNative {
+            checkNotNull(Godot.gdnative.godot_pool_color_array_remove)(it, idx)
+        }
+    }
+
+    /**
+     * Sets the size of the array. If the array is grown, reserves elements at the end of the array.
+     * If the array is shrunk, truncates the array to the new size.
+     */
+    fun resize(size: Int) {
+        callNative {
+            checkNotNull(Godot.gdnative.godot_pool_color_array_resize)(it, size)
+        }
+    }
+
+    /**
+     * Changes the color at the given index.
+     */
+    operator fun set(idx: Int, data: Color) {
+        callNative {
+            checkNotNull(Godot.gdnative.godot_pool_color_array_set)(it, idx, data.getRawMemory(this).reinterpret())
+        }
+    }
+
+    /**
+     * Returns the size of the array.
+     */
+    fun size(): Int {
+        return callNative {
+            checkNotNull(Godot.gdnative.godot_pool_color_array_size)(it)
+        }
+    }
+
+    //UTILITIES
+    operator fun plus(other: Color) {
+        this.append(other)
+    }
+
+    operator fun plus(other: PoolColorArray) {
+        this.appendArray(other)
+    }
+
+    override fun toString(): String {
+        return "PoolColorArray(${size()})"
+    }
+
+    override fun iterator(): Iterator<Color> {
+        return IndexedIterator(size(), this::get)
+    }
+
+    /**
+     * WARNING: no equals function is available in the Gdnative API for this Coretype.
+     * This methods implementation works but is not the fastest one.
+     */
+    override fun equals(other: Any?): Boolean {
+        if (other is PoolColorArray) {
+            if (other.size() != this.size()) {
+                return false
+            }
+            val iter1 = this.iterator()
+            val iter2 = other.iterator()
+            while (iter1.hasNext()) {
+                if (iter1.next() != iter2.next())
```

Comment:
this will fail if `other` is shorter than `this`, can we convert the two iterators into a `List` first then do the comparisons using list equality. i.e `thisList == otherList`. I don't think it needs to be fast, as most likely the implementation will be a placeholder for the actual gdnative implementation.

Reply:
Jus before I checked if they both have the same size so one can't be shorter than the other. 
But otherwise you are right. I guess I remade the wheel.